components/style/values/specified/position.rs

```diff
@@ -17,39 +17,104 @@ use values::specified::{LengthOrPercentage, Percentage};
 #[derive(Debug, Clone, PartialEq, Copy)]
 #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
 pub struct Position {
-    pub horizontal: LengthOrPercentage,
-    pub vertical: LengthOrPercentage,
+    pub horiz_keyword: Option<Keyword>,
+    pub horiz_position: Option<LengthOrPercentage>,
+    pub vert_keyword: Option<Keyword>,
+    pub vert_position: Option<LengthOrPercentage>,
 }
 
 impl ToCss for Position {
     fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
-        try!(self.horizontal.to_css(dest));
+        // TODO: canaltinova: We should add keywords, probably?
```

Comment:
Actually if background position is `center bottom 10px`, Position::new function sends `center` as value and it stores as `50% bottom 10px`. Keywords are storing if second keyword is explicitly defined. Otherwise it's assuming LengthOrPercentage. Is it a bad behaviour? If so, we have to check PositionComponents at Position::new whether if they are keyword or length.